### DISCUSSION TITLE
enabling rulesets

### DISCUSSION THREAD

**Discussion by aa18afr** [Security-Onion-Solutions/securityonion#11108] [General] [answered]
Hi,

I have installed Securityonion 2.3.5 recently with Snort Subscriber Ruleset. The setup works fine, but I also want to enable ET OPEN rulesets along with the current Snort rulesets. Can someone let me know how to enable them as I have gone through the documentation but could not understand how to enable them? Previously, in Securityonon 16, both rulesets were selected in the initial setup. For this version, please let me know how to do it, many thanks.

**Comment by dougburks** [comment]
Have you reviewed the Rules section of the documentation?
https://docs.securityonion.net/en/2.3/rules.html

**Reply by aa18afr** [reply]
Yes, I have reviewed it and want to enable two rulesets together. From basic installation, ETOPEN ruleset is selected but going into the file (/opt/so/saltstack/local/pillar/minions/<manager.sls>) and made changes as in the documentation to switch to new ruleset. Afterward, running: salt \* state.highstate; provides error and no TALOS ruleset is downloaded. Similar is the case if from a basic installation, I enable TALOS but later on make changes to manager.sls w.r.t to ETOPEN shows no emerging rules. Kindly let me know how to enable two rulesets together i.e., TALOS and ETOPEN, many thanks.

**Reply by dougburks** [reply]
Have you tried adding the second ruleset as shown at the bottom of that page?
https://docs.securityonion.net/en/2.3/rules.html#other

**Reply by aa18afr** [reply]
I have gone through the documentation and applied these settings as mentioned. I need to know that if I have configured security onion with Snort TALOS ruleset then how can I configure ET OPEN to run as well. As per the documentation, I need to provide a URL for ET OPEN and provided URL as url: https://rules.emergingthreats.net/open/suricata-6.0.0/emerging.rules.tar.gz

After adding url to path: /opt/so/saltstack/local/pillar/minions/<manager.sls> and running salt \* state.highstate gives error and also if so-rule-update is run, even then there is no ET OPEN ruleset is downloaded. If issue pertains with the url or I am doing anything wrong, please address, thank you.

**Reply by dougburks** [reply]
> After adding url to path: /opt/so/saltstack/local/pillar/minions/<manager.sls>

Did you actually create a file called `<manager.sls>` or are you using the existing sls file for your manager?

Are you able to share the entire `idstools` section of your manager's sls file including what you added (redacting oinkcodes as necessary)?

>  running salt * state.highstate gives error

Are you able to share the exact error?

Have you checked /opt/so/log/ for additional clues?

**Reply by aa18afr** [reply]
**Did you actually create a file called <manager.sls> or are you using the existing sls file for your manager?**
-> No, I am adding ruleset in the already present file which is placed at /opt/so/saltstack/local/pillar/minions/<manager.sls>

**Are you able to share the entire idstools section of your manager's sls file including what you added (redacting oinkcodes as necessary)?**
--> Please find the screenshot and I think I am doing something wrong in this file, please guide, thank you.

![ruleset](https://github.com/Security-Onion-Solutions/securityonion/assets/142659986/c08d2afd-b702-4058-ae19-c96468e89ac9)


**Are you able to share the exact error?**

_salt \* state.highstate error:_

![salt state error](https://github.com/Security-Onion-Solutions/securityonion/assets/142659986/2aca407b-818f-4f8a-a63e-58d9886616d2)

master _log:_

![master log](https://github.com/Security-Onion-Solutions/securityonion/assets/142659986/999898a2-4f1f-4bdb-a503-55aba64b9508)

**Reply by dougburks** [reply, marked as answer]
Looking at your manager.sls screenshot, have you tried having a single `config` section under `idstools` that contains your TALOS ruleset and oinkcode and then specifies the ET URL under `urls` as shown at https://docs.securityonion.net/en/2.3/rules.html#other?

**Reply by aa18afr** [reply]
Thanks, Doug, I tried adding the URL for ET OPEN and it worked fine for now, many thanks.
